pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -101,6 +95,24 @@ type TaskRef struct {
 	APIVersion string `json:"apiVersion,omitempty"`
 }
 
+// PipelineSource
+type PipelineSource struct {
+	Name      string    `json:"name"`
+	Type      string    `json:"type"`
+	SourceRef SourceRef `json:"sourceRef"`
+	Params    []Param   `json:"params,omitempty"`
+}
+
+// SourceRef can be used to refer to a specific instance of a Source.
+type SourceRef struct {
```

Comment:
So is `SourceRef` the interface that `Sources` must comply with? Would a `SourceRef` be expected to have the functions that the git + image types above have? e.g. `getVersion`?

Reply:
i think so :)